client/client.go

```diff
@@ -118,35 +122,24 @@ func (p *Proxy) SubscribeNewBlocks(subscriber string) (<-chan tmctypes.ResultEve
 // Tx queries for a transaction from the REST client and decodes it into a sdk.Tx
 // if the transaction exists. An error is returned if the tx doesn't exist or
 // decoding fails.
-func (p *Proxy) Tx(hash string) (sdk.TxResponse, error) {
-
-	resp, err := p.httpClient.Get(fmt.Sprintf("%s/txs/%s", p.restNode, hash))
-	if err != nil {
-		return sdk.TxResponse{}, err
-	}
-
-	defer resp.Body.Close()
-
-	bz, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return sdk.TxResponse{}, err
-	}
-	if resp.StatusCode != http.StatusOK {
-		return sdk.TxResponse{}, fmt.Errorf("response error fetching transaction with status [%d] -  %s ", resp.StatusCode, resp.Status)
-	}
-	var tx sdk.TxResponse
-	if err := p.cdc.UnmarshalJSON(bz, &tx); err != nil {
-		return sdk.TxResponse{}, err
-	}
-	return tx, nil
+func (p *Proxy) Tx(hash string) (*sdk.TxResponse, error) {
+	initClientCtx := client.Context{}.
+		WithJSONMarshaler(p.encodingConfig.Marshaler).
+		WithInterfaceRegistry(p.encodingConfig.InterfaceRegistry).
+		WithTxConfig(p.encodingConfig.TxConfig).
+		WithLegacyAmino(p.encodingConfig.Amino).
```

Comment:
So this is still using Amino?

Reply:
Its mostly just wiring, don't think we really need it but leaving it for now.